cibyl/orchestrator.py

```diff
@@ -96,37 +95,50 @@ def add_system_to_environment(self, environment,
             re_result = re.search(r'unexpected keyword argument (.*)',
                                   ex.args[0])
             if re_result:
-                raise NonSupportedSystemKey(system_name, re_result.group(1))
+                raise conf_exc.NonSupportedSystemKey(
+                    system_name, re_result.group(1))
+            re_missing_arg = re.search(r'required positional argument: (.*)',
+                                       ex.args[0])
+            if re_missing_arg:
+                raise conf_exc.MissingSystemKey(
+                    system_name, re_missing_arg.group(1))
+            raise
 
     def create_ci_environments(self) -> None:
         """Creates CI environment entities based on loaded configuration."""
-        try:
-            if self.config.data:
-                env_data = self.config.data.get('environments', {}).items()
-            else:
-                env_data = {}
+        env_data = self.config.data.get('environments', {})
+
+        if isinstance(env_data, str):
+            raise conf_exc.MissingSystems(env_data)
+        if not env_data:
+            raise conf_exc.MissingEnvironments()
 
-            for env_name, systems_dict in env_data:
+        for env_name, systems_dict in env_data.items():
+            if isinstance(systems_dict, str):
+                raise conf_exc.MissingSystemType(systems_dict, SystemType)
+            try:
```

Comment:
What does this try/except intend to catch? If the env_data is a string the above exception would make it unnecessary

Reply:
Partially I agree, but the try is around `system_dict`, not `env_data`; however, there is default value passed in `get()`, so I wonder if the `AttributeError` can occur here now?

Reply:
You're right, where I wrote env_data I meant systems_dict

Reply:
> What does this try/except intend to catch? If the env_data is a string the above exception would make it unnecessary

It tries to catch this scenario:

```
environments:
  env1
```

Reply:
But that would be caugh by the if in line 111, wouldn't it?

Reply:
> But that would be caugh by the if in line 111, wouldn't it?

different sections of the configuration file

Reply:
Ok, that makes sense, thanks!